### PR TITLE
request regular logs when previous logs are not available

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/paas 

first commit is cleanup, second is the change

### CODE REVIEW

File: plugins/kubernetes/app/models/kubernetes/deploy_executor.rb [hunk 1]
```diff
@@ -85,29 +85,45 @@ def fetch_pods(client, query)
...
-      bad_pods = release.clients.flat_map do |client, query, deploy_group|
-        bad_pods = fetch_pods(client, query).select { |p| p.restarted? || !p.live? }
-        bad_pods.map { |p| [p, client, deploy_group] }
-      end
-
-      bad_pods.each do |pod, client, deploy_group|
-        namespace = deploy_group.kubernetes_namespace
+      bad_pods(release).each do |pod, client, deploy_group|
         @output.puts "\n#{deploy_group.name} pod #{pod.name}:"
+        namespace = deploy_group.kubernetes_namespace
```
Comment (jonmoter) [inline plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: FYI, if you're getting the pod back from Kubernetes, the namespace is in the pod itself.  `pod.metadata.namespace`, i believe.

Comment (jonmoter) [inline plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: Might want to add that as a helper method to `Kubernetes::Api::Pod`

**Comment by grosser** [comment]
I think this is wrong anyways, the issue was that the pod had multiple containers :/

**Comment by grosser** [comment]
now handles multiple containers as well + added pod#namespace

**Comment by jonmoter** [comment]
`DeployExecutor` is starting to get kindof confusing to follow, and is probably due for a refactor soon. But we're still iterating on this, so I'm fine with holding off on that 'til we get the functionality more-or-less right.

👍 on 💚

**Pull request merged**